docs/Configuration.md

```diff
@@ -742,6 +742,85 @@ export default defineConfig({
 
 :::
 
+### `future` \[object]
+
+Default: `{}`
+
+Enables experimental features that may become the default behavior in future Jest releases. This allows users to opt-in to upcoming changes gradually and provide feedback.
+
+#### `future.experimental_vite` \[boolean | object]
+
+Default: `undefined`
+
+Enables experimental Vite integration for Jest. This feature allows Jest to leverage Vite's fast transformation and module resolution capabilities during test execution.
+
+When set to `true`, Jest will use Vite with default settings. You can also pass a configuration object to customize Vite's behavior:
+
+```js tab title="jest.config.js"
+const {defineConfig, withViteConfig} = require('jest-config');
+
+module.exports = defineConfig({
+  future: {
+    experimental_vite: withViteConfig({
+      configFile: './vite.config.ts',
+      mode: 'test',
+      resolve: {
+        alias: {
+          '@': '/src',
+        },
+      },
+    }),
+  },
+});
+```
+
+```ts tab title="jest.config.ts"
+import {defineConfig, withViteConfig} from 'jest-config';
+
+export default defineConfig({
+  future: {
+    experimental_vite: withViteConfig({
+      configFile: './vite.config.ts',
+      mode: 'test',
+      resolve: {
+        alias: {
+          '@': '/src',
+        },
+      },
+    }),
+  },
+});
+```
+
+Available Vite configuration options:
+
+- `configFile` \[string | false]: Path to Vite config file. Set to `false` to disable automatic config file loading.
+- `root` \[string]: Project root directory (default is the location of jest config).
+- `mode` \[string]: Mode for Vite (default: "test").
```

Comment:
I don’t see any useful usages to have this option now. We might consider adding this to later phase